spring-kafka/src/main/java/org/springframework/kafka/transaction/KafkaTransactionManager.java

```diff
@@ -0,0 +1,225 @@
+	 */
+	public KafkaTransactionManager(ProducerFactory<K, V> producerFactory) {
+		Assert.notNull(producerFactory, "The 'ProducerFactory' cannot be null");
+		setTransactionSynchronization(SYNCHRONIZATION_NEVER);
+		this.producerFactory = producerFactory;
+	}
+
+	/**
+	 * Get the producer factory.
+	 * @return the producerFactory
+	 */
+	public ProducerFactory<K, V> getProducerFactory() {
+		return this.producerFactory;
+	}
+
+	@Override
+	public Object getResourceFactory() {
+		return getProducerFactory();
+	}
+
+	@SuppressWarnings("unchecked")
+	@Override
+	protected Object doGetTransaction() {
+		KafkaTransactionObject<K, V> txObject = new KafkaTransactionObject<K, V>();
+		txObject.setResourceHolder((KafkaResourceHolder<K, V>) TransactionSynchronizationManager
+				.getResource(getProducerFactory()));
+		return txObject;
+	}
+
+	@Override
+	protected boolean isExistingTransaction(Object transaction) {
+		@SuppressWarnings("unchecked")
+		KafkaTransactionObject<K, V> txObject = (KafkaTransactionObject<K, V>) transaction;
+		return (txObject.getResourceHolder() != null);
+	}
+
+	@Override
+	protected void doBegin(Object transaction, TransactionDefinition definition) {
+		if (definition.getIsolationLevel() != TransactionDefinition.ISOLATION_DEFAULT) {
+			throw new InvalidIsolationLevelException("AMQP does not support an isolation level concept");
+		}
+		@SuppressWarnings("unchecked")
+		KafkaTransactionObject<K, V> txObject = (KafkaTransactionObject<K, V>) transaction;
+		KafkaResourceHolder<K, V> resourceHolder = null;
+		try {
+			resourceHolder = ProducerFactoryUtils.getTransactionalResourceHolder(getProducerFactory());
+			resourceHolder.getProducer().beginTransaction();
+			if (logger.isDebugEnabled()) {
+				logger.debug("Created Kafka transaction on producer [" + resourceHolder.getProducer() + "]");
+			}
+			txObject.setResourceHolder(resourceHolder);
+			txObject.getResourceHolder().setSynchronizedWithTransaction(true);
+			int timeout = determineTimeout(definition);
+			if (timeout != TransactionDefinition.TIMEOUT_DEFAULT) {
+				txObject.getResourceHolder().setTimeoutInSeconds(timeout);
+			}
+			TransactionSynchronizationManager.bindResource(getProducerFactory(), txObject.getResourceHolder());
+		}
+		catch (Exception ex) {
+			if (resourceHolder != null) {
+				ProducerFactoryUtils.releaseResources(resourceHolder);
+			}
+			throw new CannotCreateTransactionException("Could not create Kafka transaction", ex);
+		}
+	}
+
+	@Override
+	protected Object doSuspend(Object transaction) {
+		@SuppressWarnings("unchecked")
+		KafkaTransactionObject<K, V> txObject = (KafkaTransactionObject<K, V>) transaction;
+		txObject.setResourceHolder(null);
+		return TransactionSynchronizationManager.unbindResource(getProducerFactory());
+	}
+
+	@Override
+	protected void doResume(Object transaction, Object suspendedResources) {
+		@SuppressWarnings("unchecked")
+		KafkaResourceHolder<K, V> conHolder = (KafkaResourceHolder<K, V>) suspendedResources;
```

Comment:
Nit-pick, of course, but it's not about *connection*. So, just `holder` or `resourceHolder`